kernel/src/hil/digest.rs

```diff
@@ -56,13 +108,70 @@ pub trait Digest<'a, const L: usize> {
     /// error with error code ENOSUPPORT.
     fn run(&'a self, digest: &'static mut [u8; L])
         -> Result<(), (ErrorCode, &'static mut [u8; L])>;
+}
 
-    /// Clear the keys and any other sensitive data.
-    /// This won't clear the buffers provided to this API, that is up to the
-    /// user to clear.
-    fn clear_data(&self);
+/// Computes a digest (cryptographic hash) over data
```

Comment:
Over which data? Perhaps "computes a digest (cryptographic hash) over data provided through a separate trait". That way it could be DigestData or some other trait (for other use cases).